### PR TITLE
Add avoid-double-conversion flag for Cabal

### LINKED ISSUE #80: putting double-conversion behind a build flag?
The double-conversion dependency seems quite heavy and there are many use-cases that don't need formatting of floating point.

So I'd like to suggest the idea of putting the usage of double-conversion under a Cabal flag (which can still default to true)
to allow easily building formatting without double-conversion. How do you think?

### PR DESCRIPTION
**Pull request by ysangkok**
Fixes #80. I had to adjust the existing non-double-conversion code since it was failing tests.

### CODE REVIEW

**Comment by ysangkok** [comment]
ping @AJChapman

File: formatting.cabal [hunk 1]
```diff
@@ -48,6 +48,11 @@ common deps
     ghc-options:       -Wno-prepositive-qualified-module
                        -Wno-missing-safe-haskell-mode
 
+flag avoid-double-conversion
```
Comment (juhp) [inline formatting.cabal:51]: Minor point, but I think `no-double-conversion` or `disable-double-conversion` would be a better flag name

**Comment by ysangkok** [comment]
Thanks for the merge! Looking forward to the release of v7.2.0.

File: src/Data/Text/Format.hs [hunk 2]
```diff
@@ -53,11 +54,11 @@ fixed :: (Real a) =>
          Int
       -- ^ Number of digits of precision after the decimal.
       -> a -> Builder
-#ifdef ghcjs_HOST_OS
+#ifndef MIN_VERSION_double_conversion
 fixed decs = fromString . toFixed . realToFrac
   where
     toFixed :: Double -> String
-    toFixed = printf ("%f." ++ show decs)
+    toFixed dbl = showFFloat (Just decs) dbl ""
```
Comment (Bodigrim) [inline src/Data/Text/Format.hs:61]: Why not use https://hackage.haskell.org/package/text-2.0.1/docs/Data-Text-Lazy-Builder-RealFloat.html#v:realFloat directly?

Comment (ysangkok) [reply src/Data/Text/Format.hs:61]: `realFloat` uses scientific notation, but this function is supposed to use "normal notation" as its Haddock says.

Comment (Bodigrim) [reply src/Data/Text/Format.hs:61]: Would `formatRealFloat Fixed Nothing` do the trick?

Comment (ysangkok) [reply src/Data/Text/Format.hs:61]: Yes, I think that would work, looking at the definition of `showFFloat`. But why is it preferable?

Comment (Bodigrim) [reply src/Data/Text/Format.hs:61]: It's a more direct approach to create a `Builder` directly, without `String` overhead.

Comment (ysangkok) [reply src/Data/Text/Format.hs:61]: Submitted #83

**Comment by juhp** [comment]
Just noting that the flag is enabled in Stackage Nightly

**Pull request merged**